.devcontainer/post-create-command.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -e
+
+# Add user's fork as a remote
+GIT_USER=$(git config user.name)
+if ! git remote | grep -q "fork"; then
+    git remote add fork https://github.com/${GIT_USER}/flask
+fi
+
+# Upgrade pip and setuptools
+python -m pip install --upgrade pip setuptools
+
+# Install the development dependencies, then install Flask in editable mode
+pip install -r requirements/dev.txt && pip install -e .
```

Comment:
This needs to install into a virtualenv. Installing to the system Python, even in a container, can lead to unexpected issues.

Reply:
Just so I understand clearly, since I'm not familiar with Python development, are users expected to do all their development and run tests within a venv?

I'm trying to figure out the cleanest way for to activate a venv for users once the devcontainer has finished installing the dependencies. So far, the options are to:
1. Simply prompt users to activate their venv in the welcome text (I think this is the easiest route, but users might forget to do this since the text only shows after initial creation)
2. Use a `postAttachCommand` in the devcontainer to open a new terminal with the venv activated
3. Configure the .bashrc and other popular terminals to automatically run the activation command when opened (this has the best coverage but requires a bit more work)

I'm interested to see what you think is the best option for contributors.

Reply:
Yes, Python projects use virtualenvs both during development and deployment. You might find this article about activating a virtualenv in a container useful: https://pythonspeed.com/articles/activate-virtualenv-dockerfile/. Basically all activating actually does is set the `VIRTUAL_ENV` env var to the base path, and add `$VIRTUAL_ENV/bin` to `PATH`.